subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/java/JavaCleanAssemblePerformanceTest.groovy

```diff
@@ -17,37 +17,22 @@
 package org.gradle.performance.regression.java
 
 import org.gradle.performance.AbstractCrossVersionPerformanceTest
-import spock.lang.Unroll
-
-import static org.gradle.performance.generator.JavaTestProject.LARGE_JAVA_MULTI_PROJECT
-import static org.gradle.performance.generator.JavaTestProject.LARGE_MONOLITHIC_JAVA_PROJECT
-import static org.gradle.performance.generator.JavaTestProject.MEDIUM_JAVA_COMPOSITE_BUILD
-import static org.gradle.performance.generator.JavaTestProject.MEDIUM_JAVA_PREDEFINED_COMPOSITE_BUILD
 
 class JavaCleanAssemblePerformanceTest extends AbstractCrossVersionPerformanceTest {
 
-    @Unroll
-    def "clean assemble on #testProject"() {
+    def "clean assemble"() {
         given:
-        runner.testProject = testProject
-        runner.gradleOpts = ["-Xms${testProject.daemonMemory}", "-Xmx${testProject.daemonMemory}"]
-        runner.warmUpRuns = warmUpRuns
-        runner.runs = runs
+        runner.gradleOpts = runner.projectMemoryOptions
+        runner.warmUpRuns = 2
+        runner.runs = 6
         runner.tasksToRun = ["clean", "assemble"]
         runner.targetVersions = ["6.7-20200824220048+0000"]
-        runner.minimumBaseVersion = minimumBaseVersion
+        runner.minimumBaseVersion = runner.testProject.contains("Composite") ? "4.0" : null
```

Comment:
This sounds like something that would be better modeled via something like `testProject.requiresComposite`, or maybe set automatically from `testProject.minimumBaseVersion`.

Reply:
This should be handled as part of https://github.com/gradle/gradle/issues/14597 as well I think.